invenio_communities/members/records/models.py

```diff
@@ -0,0 +1,36 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2022 Northwestern University.
+#
+# Invenio-Communities is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""Member Model."""
+
+from invenio_db import db
+from invenio_accounts.models import User
+from invenio_records.models import RecordMetadataBase
+from sqlalchemy_utils.types import UUIDType
+
+from ...communities.records.models import CommunityMetadata
+
+
+class MemberModel(db.Model, RecordMetadataBase):
```

Comment:
I don't think we should use a record with json column as table. I think it should look like:


- ``user_id`` - Foreign key to the user
- ``group_id`` - Foreign key to the role
- ``community_id`` - Foreign key to the community
- ``role`` - Role identifier (str) e.g. ``owner``, ``manager``
- ``visibility`` - Boolean (true=public, false=hidden)
- ``created`` - Creation timestamp (via ``Timestamp`` model from SQLAlchemy-Utils)
- ``updated`` - Modification timestamp. (via ``Timestamp`` model from SQLAlchemy-Utils)

We need indices on ``user_id``, ``group_id`` (select all memberships for a user), and ``community_id`` (select all memberships for a community.

Example of a query we have to do:

```sql
SELECT DISTINCT 
    community_id, role
FROM 
    community_members 
WHERE 
    user_id=<user_id> OR 
    group_id IN [<group_id>, <group_id>, ...,]
```

Also, possibly we should see if we could add a check constraint to ensure only group or user is set and not both.

Reply:
I see, might as well not inherit `RecordMetadataBase` :+1:

Reply:
Unfortunately, removing RecordMetadataBase has many unintended consequences: invenio-records/invenio_records/api.py assumes that the underlying model has all the fields of a RecordMetadataBase. (uses keyword arguments in `create`). Overriding the api.py Record is tricky for me because I don't know if most of it should be kept / roles of hte different signals/pre/post extensions. 

I will keep the inheritance but disable the field if I can.